src/kokkos/test/kokkos/AtomicPairCounter_t.cc

```diff
@@ -0,0 +1,81 @@
+#include "KokkosCore/kokkosConfigCommon.h"
+#include "KokkosCore/kokkosConfig.h"
+
+#include <iostream>
+// dirty, but works
+#include "KokkosCore/AtomicPairCounter.h"
+
+typedef Kokkos::TeamPolicy<KokkosExecSpace> team_policy;
+typedef Kokkos::TeamPolicy<KokkosExecSpace>::member_type  member_type;
+
+void test(){
+
+  Kokkos::View<AtomicPairCounter*,KokkosExecSpace> dc_d("dc_d",1);
+  Kokkos::View<AtomicPairCounter*,KokkosExecSpace>::HostMirror dc_h("dc_h",1);
+
+  std::cout << "size " << sizeof(AtomicPairCounter) << std::endl;
+
+  constexpr uint32_t N = 20000;
+  constexpr uint32_t M = N * 6;
+
+  Kokkos::View<uint32_t*,KokkosExecSpace> n_d("n_d",N);
+  Kokkos::View<uint32_t*,KokkosExecSpace> m_d("m_d",M);
+
+  const uint32_t n = 10000;
+#ifdef KOKKOS_BACKEND_CUDA
+  auto league_size = n/10;
+#elif KOKKOS_BACKEND_SERIAL
+  auto league_size = n*2;
+#else
+  auto league_size = n/2;
+#endif
```

Comment:
I'm a bit unhappy about having to do this (especially with `#ifdef`). Do I understand correctly that both `league_size` and `team_size` can not be `Kokkos::AUTO()`? Maybe this calls for some abstraction to hide the `#ifdef` from user code (to be figured out later).

Reply:
We don't have to do this. I'll remove this. Mainly we just need to ensure there are enough thread executions to reach `10000` in all cases. However, since I cannot know what `Kokkos::AUTO()` will be, I can only ensure this by making `league_size = n` for all cases. This means for parallel cases we sorely over-execute the work.

Reply:
It really raises the question of whether or not I should use `Kokkos::AUTO()` since I cannot predict how much work it will produce. If I need to execute my `parallel_for` N times, and I can set the `league_size` but the `team_size` will be set to some arbitrary value, how do I ensure the loop executes N times?